rpmlb/builder/base.py

```diff
@@ -11,6 +12,17 @@
 LOG = logging.getLogger(__name__)
 
 
+#: Regular expression for finding macro definitions
+MACRO_REGEX = re.compile(
+    r'''^
+    %global\s+                   # beginning of definition
+    (?P<name>[^\s]+)\s+          # one-word name
+    (?P<value>(?:.|(?<=\\)\n)+)  # value including spaces and escaped newlines
+    $''',
+    flags=re.MULTILINE | re.VERBOSE,
+)
```

Comment:
Because formally, RPM macro bodies are allowed to contain white space and are allowed to span multiple lines, if the newlines are escaped: http://rpm.org/user_doc/macros.html

I wanted to make sure we do not left half of the body behind when replacing, should we ever need to replace a multi-line macro.